src/Peachpie.Library/DateTime/DateTimeZone.cs

```diff
@@ -60,10 +63,24 @@ public void __construct(Context ctx, string timezone_name)
 
                 if (_timezone == null)
                 {
-                    // TODO: an offset value (+0200)
-
-                    //PhpException.Throw(PhpError.Notice, LibResources.GetString("unknown_timezone", timezone_name));
-                    throw new Spl.InvalidArgumentException();
+                    var m = TimeZoneOffsetRegex.Match(timezone_name);
```

Comment:
I just moved the code to the helper class and made the RegEx lazily initialized
https://github.com/peachpiecompiler/peachpie/commit/9d3ff3f9605fe894dd2dfb92d2141b13b3f92389
Thanks!